test/governance/Governor.test.js

```diff
@@ -297,6 +306,73 @@ contract('Governor', function (accounts) {
           });
         });
 
+        describe('on vote by signature', function () {
+          beforeEach(async function () {
+            this.voterBySig = Wallet.generate();
+            this.voterBySig.address = web3.utils.toChecksumAddress(this.voterBySig.getAddressString());
+
+            this.signature = (contract, message) =>
+              getDomain(contract)
+                .then(domain => ({
+                  primaryType: 'Ballot',
+                  types: {
+                    EIP712Domain: domainType(domain),
+                    Ballot: [
+                      { name: 'proposalId', type: 'uint256' },
+                      { name: 'support', type: 'uint8' },
+                      { name: 'voter', type: 'address' },
+                      { name: 'nonce', type: 'uint256' },
+                    ],
+                  },
+                  domain,
+                  message,
+                }))
+                .then(data => ethSigUtil.signTypedMessage(this.voterBySig.getPrivateKey(), { data }))
+                .then(fromRpcSig);
+
+            await this.token.delegate(this.voterBySig.address, { from: voter1 });
+
+            // Run proposal
+            await this.helper.propose();
+            await this.helper.waitForSnapshot();
+          });
+
+          it('if signature does not match signer', async function () {
+            const nonce = await this.mock.nonces(this.voterBySig.address);
+
+            expectRevertCustomError(
+              this.helper.vote({
+                support: Enums.VoteType.For,
+                voter: this.voterBySig.address,
+                nonce,
+                signature: (...params) => {
+                  const sig = this.signature(...params);
+                  sig[69] ^= 0xff;
+                  return sig;
+                },
+              }),
+              'GovernorInvalidSigner',
+              [],
```

Comment:
Doesn't this error have arguments?

Reply:
I realized this is an error and the tests were not working properly because they're missing an `await`. I'll mark as draft before I fix that

Reply:
Fixed